### PR TITLE
IN-4608 - Kafka monitoring docs adjustments

### PR DESCRIPTION
**Pull request by gr0**
Hi guys, this time similar changes for Kafka monitoring. I think I already incorporated all the wording/suggestion changes from the Solr/SolrCloud section, so hopefully will be easier to read :)

### CODE REVIEW

File: docs/integration/kafka.md [hunk 1]
```diff
@@ -1,8 +1,191 @@
 title: Kafka Monitoring Integration
 description: Monitor Kafka metrics for brokers, producers, and consumers, consumer lag and offset monitoring by consumer group, topic, or partition, and more. Our cloud and on-premises tools provide out of box Kafka graphs, reports and custom dashboards with built-in anomaly detection, threshold, and heartbeat alerts as well as easy chatops integrations
 
+
+Sematext offers simple Kafka monitoring agent written in Java and Golang with minimal CPU and memory overhead. It's easy to install and require no changes in the Kafka source code or your application's source code.
```
Comment (adnanrahic) [inline docs/integration/kafka.md]: It's easy to install and **requires** * no changes in the Kafka source code or your application's source code.

File: docs/integration/kafka.md [hunk 2]
```diff
@@ -1,8 +1,191 @@
...
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-broker:default
+```
+
+**You need to restart your Kafka broker after the changes above.**
+
+To get the full view over some metrics - like the consumer lag, you also need to install monitoring on each of your Kafka producers and consumers. Here is how you can do that.
+
+### Monitoring Producers
+
+To have the full visibility into the whole Kafka pipeline it is crucial to monitor your Kafka producers as well. If your are using Java or Scala as the language of your choice for the producers implementation you need to install the Kafka Monitoring Agent on each host working as Kafka producer by running the following command (i.e. for CentOS):
```
Comment (adnanrahic) [inline docs/integration/kafka.md]: To have (the - delete it)* visibility into the **entire** * Kafka pipeline **it's** crucial to monitor your Kafka producers as well. If **you're** * using Java or Scala as **your language of choice for the producers'** implementation you need to install the Kafka Monitoring Agent on each host working as **a** Kafka producer by running the following command (i.e. for CentOS):

File: docs/integration/kafka.md [hunk 3]
```diff
@@ -1,8 +1,191 @@
...
+
+```bash
+sudo wget https://pub-repo.sematext.com/centos/sematext.repo -O /etc/yum.repos.d/sematext.repo
+sudo yum clean all
+sudo yum install spm-client
+``` 
+
+After that, run following command to setup Kafka producer monitoring:
```
Comment (adnanrahic) [inline docs/integration/kafka.md]: After that, run **the** * following command to setup Kafka producer monitoring:

File: docs/integration/kafka.md (hunk 1 above)
Comment (gr0) [reply docs/integration/kafka.md]: Actually @otisg mentioned that it should be *required*.

File: docs/integration/kafka.md (hunk 2 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 3 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 1 above)
Comment (adnanrahic) [reply docs/integration/kafka.md]: Here's another variation:

> Sematext has a simple Kafka monitoring Agent written in Java and Go with minimal CPU and memory overhead. It's easy to install and doesn't require any changes to the Kafka source code or your application's source code.

Comment (otisg) [reply docs/integration/kafka.md]: Oh, if I said "required", I made a typo.  I meant requires :)
+1 for Adnan's variant, even cleaner.

File: docs/integration/kafka.md [hunk 4]
```diff
@@ -1,8 +1,191 @@
...
+    --monitoring-token <your-monitoring-token-goes-here>   \
+    --app-type kafka  \
+    --app-subtype kafka-broker  \
+    --agent-type javaagent  \
+    --infra-token <your-infra-token-goes-here>
+```
+
+Keep in mind that your need to provide the Monitoring token and Infra token. They are both provided in the [installation instructions](https://apps.sematext.com/ui/howto/Kafka/overview) for your Kafka App. 
+
+Finally, the last thing that needs to be done is adjusting the ```$KAFKA_HOME/bin/kafka-server-start.sh``` file and add the following section to the ```KAFKA_JMX_OPTS```:
```
Comment (otisg) [inline docs/integration/kafka.md]: Finally, the last..... -- Finally means "the last", so .... :)

File: docs/integration/kafka.md [hunk 5]
```diff
@@ -1,8 +1,191 @@
...
+```bash
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-broker:default
+```
+
+**You need to restart your Kafka broker after the changes above.**
+
+To get the full view over some metrics - like the consumer lag, you also need to install monitoring on each of your Kafka producers and consumers. Here is how you can do that.
```
Comment (otisg) [inline docs/integration/kafka.md]: To get the full view over some metrics - like the consumer lag,  you also need to install monitoring => To see the complete picture of Kafka performance install the monitoring agent

File: docs/integration/kafka.md [hunk 6]
```diff
@@ -1,8 +1,191 @@
...
+
+### Monitoring Producers
+
+To have the full visibility into the entire Kafka pipeline it's crucial to monitor your Kafka producers as well. If you're using Java or Scala as the language of choice for the producers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka producer by running the following command (i.e. for CentOS):
```
Comment (otisg) [inline docs/integration/kafka.md]: i.e. => e.g.

File: docs/integration/kafka.md [hunk 7]
```diff
@@ -1,8 +1,191 @@
...
+Once that is done you need to add the following options to the JVM start-up properties:
+```bash
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-producer:default
+```
+
+**You need to restart your Kafka producer after the changes above.**
+
+### Monitoring Consumers
+
+Monitoring your consumers is crucial to have the visibility into the topic lag, which can help you quickly identify issues with your pipeline. If you're using Java or Scala as the language of choice for the consumers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka consumer by running the following command (i.e. for CentOS):
```
Comment (otisg) [inline docs/integration/kafka.md]: topic lag => consumer lag
i.e. => e.g.

File: docs/integration/kafka.md [hunk 8]
```diff
@@ -1,8 +1,191 @@
...
+```bash
+sudo bash /opt/spm/bin/setup-sematext  \
+    --monitoring-token <your-monitoring-token-goes-here>   \
+    --app-type kafka  \
+    --app-subtype kafka-consumer  \
+    --agent-type javaagent  \
+    --infra-token <your-infra-token-goes-here>
+```
+
+Once that is done you need to add the following options to the JVM start-up properties:
```
Comment (otisg) [inline docs/integration/kafka.md]: Once that is done you need to add  => Once that is done add

File: docs/integration/kafka.md [hunk 9]
```diff
@@ -1,8 +1,191 @@
...
+- Connection count, connection create rate, connection close rate, network I/O rate
+- Consumed rate, records per request, fetch latency, fetch rate, bytes consumed rate, average fetch size, throttle maximum time
+- Assigned partitions, heartbeat maximum response time, heartbeat rate, join time and maximum join time, sync time and maximum sync time, join rate, syncs rate
+- Nodes bytes IN rate, node bytes OUT rate, request latency and max latency, request rate, responses rate, requests size and maximum size
+
+![](https://sematext.com/wp-content/uploads/2019/05/d_kafka_consumer_lag.png)
+
+## Troubleshooting
+
+If you are having issues with Sematext Monitoring, i.e. not seeing SolrCloud metrics, you can create a diagnostics package on any affected machines where SolrCloud Monitoring Agent installed by running:
```
Comment (otisg) [inline docs/integration/kafka.md]: We can be DRY...
Oh, and there is a mention of SolrCloud here.

File: docs/integration/kafka.md [hunk 10]
```diff
@@ -1,8 +1,191 @@
...
+
+```bash
+sudo bash /opt/spm/bin/spm-client-diagnostics.sh
+```
+
+The resulting package will contain all relevant info needed for our investigation. You can send it, along with a short description of your problem, to support@sematext.com or contact us through the chat in the bottom right.
+
+## Troubleshooting
```
Comment (otisg) [inline docs/integration/kafka.md]: Second Troubleshooting section?

File: docs/integration/kafka.md (hunk 1 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 4 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 5 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 6 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 7 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 9 above)
Comment (gr0) [reply docs/integration/kafka.md]: Yeah, my fault copy+paste issue :) Adjusted.

File: docs/integration/kafka.md (hunk 10 above)
Comment (gr0) [reply docs/integration/kafka.md]: Yeah, my fault copy+paste issue :) Adjusted.

File: docs/integration/kafka.md [hunk 11]
```diff
@@ -1,8 +1,181 @@
...
+
+To have the full visibility into the entire Kafka pipeline it's crucial to monitor your Kafka producers as well. If you're using Java or Scala as the language of choice for the producers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka producer by running the following command (e.g. for CentOS):
+
+```bash
+sudo wget https://pub-repo.sematext.com/centos/sematext.repo -O /etc/yum.repos.d/sematext.repo
+sudo yum clean all
+sudo yum install spm-client
+``` 
+
+After that, run the following command to setup Kafka producer monitoring:
```
Comment (adnanrahic) [inline docs/integration/kafka.md]: setup -> set up

File: docs/integration/kafka.md [hunk 12]
```diff
@@ -1,8 +1,181 @@
...
+Once that is done you need to add the following options to the JVM start-up properties:
+```bash
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-producer:default
+```
+
+**You need to restart your Kafka producer after the changes above.**
+
+### Monitoring Consumers
+
+Monitoring your consumers is crucial to have the visibility into the topic lag, which can help you quickly identify issues with your pipeline. If you're using Java or Scala as the language of choice for the consumers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka consumer by running the following command (e.g. for CentOS):
```
Comment (adnanrahic) [inline docs/integration/kafka.md]: Monitoring your consumers is crucial to have the visibility into the topic lag > Monitoring your consumers is crucial to have visibility into topic lag

File: docs/integration/kafka.md [hunk 13]
```diff
@@ -1,8 +1,181 @@
...
+
+```bash
+sudo wget https://pub-repo.sematext.com/centos/sematext.repo -O /etc/yum.repos.d/sematext.repo
+sudo yum clean all
+sudo yum install spm-client
+``` 
+
+After that, run the following command to setup Kafka consumer monitoring:
```
Comment (adnanrahic) [inline docs/integration/kafka.md]: setup -> set up

File: docs/integration/kafka.md (hunk 11 above)
Comment (gr0) [reply docs/integration/kafka.md]: Done.

File: docs/integration/kafka.md (hunk 12 above)
Comment (gr0) [reply docs/integration/kafka.md]: Done.

File: docs/integration/kafka.md (hunk 13 above)
Comment (gr0) [reply docs/integration/kafka.md]: Done.

File: docs/integration/kafka.md (hunk 8 above)
Comment (gr0) [reply docs/integration/kafka.md]: Done.

File: docs/integration/kafka.md [hunk 14]
```diff
@@ -1,8 +1,181 @@
 title: Kafka Monitoring Integration
 description: Monitor Kafka metrics for brokers, producers, and consumers, consumer lag and offset monitoring by consumer group, topic, or partition, and more. Our cloud and on-premises tools provide out of box Kafka graphs, reports and custom dashboards with built-in anomaly detection, threshold, and heartbeat alerts as well as easy chatops integrations
 
+
+Sematext has a simple Kafka monitoring Agent written in Java and Go with minimal CPU and memory overhead. It's easy to install and doesn't require any changes to the Kafka source code or your application's source code.
+
+## Sematext Kafka Monitoring Agent 
+This lightweight, open-source [Java Monitoring Agent](https://github.com/sematext/sematext-agent-java) collects Kafka performance metrics and sends them to Sematext. It comes packaged with a Golang-based agent responsible for Operating System level metrics like network, disk I/O, and more. The Kafka Monitoring Agent can be installed with RPM/DEB package manager on any host running Linux or in a containerized environment using ```sematext/spm-client```.
```
Comment (otisg) [inline docs/integration/kafka.md]: Hey @gr0 @adnanrahic @bsmid when you read this "....Java Monitoring Agent" I always get this bad feeling that this sounds like it's an agent for monitoring Java/JVM apps.  The language of implementation is not really important.  We only use it do differentiate between agents written in different languages, but that is not something users (need to) care/think about.  For example, in this case, we wouldn't lose anything by just saying "This lightweight, open-source Monitoring Agent....". Plus, look how we later refer to it as "Kafka Monitoring Agent", which is really more relevant.  It's more important to call it by what it's monitoring than what language it is written in.

File: docs/integration/kafka.md [hunk 15]
```diff
@@ -1,8 +1,181 @@
...
+Once that is done you need to add the following options to the JVM start-up properties:
+```bash
+-Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>:kafka-producer:default
+```
+
+**You need to restart your Kafka producer after the changes above.**
+
+### Monitoring Consumers
+
+Monitoring your consumers is crucial to have visibility into topic lag, which can help you quickly identify issues with your pipeline. If you're using Java or Scala as the language of choice for the consumers' implementation you need to install the Kafka Monitoring Agent on each host working as a Kafka consumer by running the following command (e.g. for CentOS):
```
Comment (otisg) [inline docs/integration/kafka.md]: topic lag or consumer lag?

File: docs/integration/kafka.md [hunk 16]
```diff
@@ -1,8 +1,181 @@
...
+
+### Kafka Consumer
+
+- Consumer lag
+- Fetcher responses, bytes, responses bytes
+- I/O time, I/O ratio, I/O wait time, I/O wait ratio
+- Connection count, connection create rate, connection close rate, network I/O rate
+- Consumed rate, records per request, fetch latency, fetch rate, bytes consumed rate, average fetch size, throttle maximum time
+- Assigned partitions, heartbeat maximum response time, heartbeat rate, join time and maximum join time, sync time and maximum sync time, join rate, syncs rate
+- Nodes bytes IN rate, node bytes OUT rate, request latency and max latency, request rate, responses rate, requests size and maximum size
```
Comment (otisg) [inline docs/integration/kafka.md]: Why are IN and OUT capitalized?

responses rate => response rate
requests size => request size

File: docs/integration/kafka.md [hunk 17]
```diff
@@ -1,8 +1,181 @@
...
+
+If you are having issues with Sematext Monitoring, i.e. not seeing Kafka metrics, you can create a diagnostics package on any affected machines where Kafka Monitoring Agent installed by running:
+
+```bash
+sudo bash /opt/spm/bin/spm-client-diagnostics.sh
+```
+
+The resulting package will contain all relevant info needed for our investigation. You can send it, along with a short description of your problem, to support@sematext.com or contact us through the chat in the bottom right.
+
+For more troubleshooting information please look at [Troubleshooting](/monitoring/spm-faq/#troubleshooting) section.
```
Comment (otisg) [inline docs/integration/kafka.md]: No need for "please" in docs.

Comment (gr0) [reply docs/integration/kafka.md]: My manner of adding that "please". Removed.

File: docs/integration/kafka.md (hunk 16 above)
Comment (gr0) [reply docs/integration/kafka.md]: Adjusted.

File: docs/integration/kafka.md (hunk 15 above)
Comment (gr0) [reply docs/integration/kafka.md]: Yes, consumer lag, not topic lag, you are right. The first version was good, this is not, adjusted. Good catch, thanks!

File: docs/integration/kafka.md (hunk 14 above)
Comment (gr0) [reply docs/integration/kafka.md]: Do you think that changing the `Java Monitoring Agent` to `Kafka Monitoring Agent` sounds better to `Monitoring Agent` itself? I like the `Kafka Monitoring Agent` better, but I'm not sure.

Comment (otisg) [reply docs/integration/kafka.md]: 👍 for <integration> Monitoring Agent

Comment (adnanrahic) [reply docs/integration/kafka.md]: Agree. Just using **Monitoring Agent** is optimal here.

**Pull request merged**